mods/sethome/init.lua

```diff
@@ -2,64 +2,59 @@ local homes_file = minetest.get_worldpath() .. "/homes"
 local homepos = {}
 
 local function loadhomes()
-    local input = io.open(homes_file, "r")
-    if input then
+	local input = io.open(homes_file, "r")
```

Comment:
Second return value is the error message, which should either be propagated or at least printed to the log, if input is nil.